server/src/main/java/org/elasticsearch/script/VectorScoreScriptUtils.java

```diff
@@ -470,7 +497,11 @@ public ByteCosineSimilarity(ScoreScript scoreScript, DenseVectorDocValuesField f
 
         public double cosineSimilarity() {
             setNextVector();
-            return field.get().cosineSimilarity(queryVector, qvMagnitude);
+            if (floatQueryVector != null) {
+                return field.get().cosineSimilarity(floatQueryVector);
```

Comment:
```suggestion
                return field.get().cosineSimilarity(floatQueryVector, false);
```
We should normalize the floating point query and pass a false here.